src/state.py

```diff
@@ -0,0 +1,60 @@
+from abc import ABC, abstractmethod
+
+
+"""
+Game state could be changed from the <Context> object as well as from any <BaseState> descendant
+"""
+
+
+class Context(ABC):
+    _state = None
+
+    def __init__(self, state):
+        self.transition_to(state)
+
+    def transition_to(self, state):
+        print(f"[CONTEXT] Transition to {type(state).__name__}")
+        self._state = state
+        self._state.context = self
+
+    def get_active_state(self):
+        self._state.whoami()
+
+
+class BaseState(ABC):
+    @property
+    def context(self):
```

Comment:
Is it really needed ?
This leads to cyclic links

Reply:
Let's decide.
In case if we'll need to change the state ONLY from the <Context>, e.g. only context will handle state transitions- then no, it's not necessary.
If we'll need to change the state from the <State> obj - we need to keep the link from the state to context.